src/timeman.cpp

```diff
@@ -94,6 +94,14 @@ void TimeManagement::init(Search::LimitsType& limits, Color us, int ply) {
   optimumTime = TimePoint(optScale * timeLeft);
   maximumTime = TimePoint(std::min(0.8 * limits.time[us] - moveOverhead, maxScale * optimumTime));
 
+  if (Stockfish::Search::Limits.use_time_management())
+  {
+      int strength = std::clamp( int(std::log(optimumTime * Threads.size() / 10) * 60), 1, 1024);
```

Comment:
if optimimum time is < 10 this will lead to a std::log(0), which might be a problem. A simple max(1, ...) would fix this, I think.
second, why do we need a clamp on both strength and tempoNNUE, probably that could be rewritten in identical form with one clamp only, as it is linear.

Can you verify something like this is equivalent:
```c++
int strength = std::log(std::max(1, optimumTime * Threads.size() / 10)) * 60;
tempoNNUE = std::clamp((strength + 264) / 24, 18, 30);
```

Reply:
Good spot with the log - I will modify it, apologies for not thinking about that once the code looked usable.
Yes, the clamp on strength should not be necessary, I used it for clarity of thinking when first writing the code. I was thinking the upper bound could be a lot higher, say 1,000,000 or so, to allow for really strong hw, but in practice that range is not used for this tempo change. I will do some checks ...

Reply:
I have updated the code. I included a cast to prevent issues with int and different unsigned long types on the max().